prover/src/lib.rs

```diff
@@ -496,44 +480,53 @@ pub trait Prover {
         // - interpolate the trace into a polynomial in coefficient form
         // - "break" the polynomial into a set of column polynomials each of degree equal to
         //   trace_length - 1
-        #[cfg(feature = "std")]
-        let now = Instant::now();
-        let composition_poly =
-            CompositionPoly::new(composition_poly_trace, domain, num_trace_poly_columns);
-        #[cfg(feature = "std")]
-        debug!(
-            "Converted constraint evaluations into {} composition polynomial columns of degree {} in {} ms",
-            composition_poly.num_columns(),
-            composition_poly.column_degree(),
-            now.elapsed().as_millis()
-        );
+        let composition_poly = info_span!(
+            "Converted constraint evaluations into composition polynomial columns",
+            columns_number = field::Empty,
+            degree = field::Empty
+        )
+        .in_scope(|| {
+            let composition_poly =
+                CompositionPoly::new(composition_poly_trace, domain, num_trace_poly_columns);
+
+            tracing::Span::current().record("columns_number", composition_poly.num_columns());
+            tracing::Span::current().record("degree", composition_poly.column_degree());
+
+            composition_poly
+        });
 
         // then, evaluate composition polynomial columns over the LDE domain
-        #[cfg(feature = "std")]
-        let now = Instant::now();
-        let composed_evaluations = RowMatrix::evaluate_polys_over::<DEFAULT_SEGMENT_WIDTH>(
-            composition_poly.data(),
-            domain,
-        );
-        #[cfg(feature = "std")]
-        debug!(
-            "Evaluated {} composition polynomial columns over LDE domain (2^{} elements) in {} ms",
-            composed_evaluations.num_cols(),
-            composed_evaluations.num_rows().ilog2(),
-            now.elapsed().as_millis()
-        );
+        let composed_evaluations = info_span!(
+            "Evaluated composition polynomial columns over LDE domain",
+            columns_number = field::Empty,
+            domain_size = field::Empty
+        )
+        .in_scope(|| {
```

Comment:
Let's make the following changes here:

1. Change the name to "evaluate_composition_poly_columns".
2. For `num_columns` field use `num_constraint_composition_columns`.
3. For `domain_size` field let's use `domain.lde_domain_size()`.
4. Assert that the relevant fields are set correctly.